hoot-core/src/main/cpp/hoot/core/OsmMap.cpp

```diff
@@ -933,6 +883,13 @@ bool OsmMap::validate(bool strict) const
 
 void OsmMap::visitRo(ElementVisitor& visitor) const
 {
+  visitNodesRo(visitor);
+  visitWaysRo(visitor);
+  visitRelationsRo(visitor);
+}
+
+void OsmMap::visitNodesRo(ElementVisitor& visitor) const
```

Comment:
I was like, it's only three small methods... but now I'm thinking I can collapse this, not as a template, but if I use the ElementType class. I'll see what I can come up with.